examples/thompson_sampling.py

```diff
@@ -0,0 +1,302 @@
+    # set uninformative log-normal priors on our kernel hyperparameters
+    var = numpyro.sample("var", dist.LogNormal(0.0, 1.0))
+    length = numpyro.sample("length", dist.LogNormal(0.0, 1.0))
+
+    # compute kernel
+    k = kernel(X, X, var, length)
+
+    # sample Y according to the standard gaussian process formula
+    numpyro.sample(
+        "Y",
+        dist.MultivariateNormal(loc=jnp.zeros(X.shape[0]), covariance_matrix=k),
+        obs=Y,
+    )
+
+
+class GP:
+    """ Adapted to numpyro from https://gdmarmerola.github.io/ts-for-bayesian-optim/ """
+
+    def __init__(self, kernel=matern32_kernel):
+        self.kernel = kernel
+        self.kernel_params = None
+
+    def fit(self, X, Y, rng_key, n_step):
+        self.X_train = X
+
+        # store moments of training y (to normalize)
+        self.y_mean = jnp.mean(Y)
+        self.y_std = jnp.std(Y)
+
+        # normalize y
+        Y = (Y - self.y_mean) / self.y_std
+
+        # setup optimizer and SVI
+        optim = numpyro.optim.Adam(step_size=0.005)
+
+        svi = SVI(
+            model,
+            guide=AutoDelta(model),
+            optim=optim,
+            loss=Trace_ELBO(),
+            X=X,
+            Y=Y,
+        )
+
+        params, _ = svi.run(rng_key, n_step)
+
+        # get kernel parameters from guide with proper names
+        self.kernel_params = svi.guide.median(params)
+
+        # store inverted prior covariance
+        self.K_xx_inv = jnp.linalg.inv(self.kernel(X, X, **self.kernel_params))
+
+        # store inverted prior covariance multiplied by y
+        self.alpha = self.K_xx_inv @ Y
+
+        return self.kernel_params
+
+    # do GP prediction for a given set of hyperparameters. this makes use of the well-known
+    # formula for gaussian process predictions
+    def predict(self, X, return_std=False):
+        # compute kernels between train and test data, etc.
+        k_pp = self.kernel(X, X, **self.kernel_params)
+        k_pX = self.kernel(X, self.X_train, **self.kernel_params, jitter=0.0)
+
+        # compute posterior covariance
+        K = k_pp - k_pX @ self.K_xx_inv @ jnp.transpose(k_pX)
+
+        # compute posterior mean
+        mean = jnp.matmul(k_pX, self.alpha)
+
+        # we return both the mean function and the standard deviation
+        if return_std:
+            return (
+                (mean * self.y_std) + self.y_mean,
+                jnp.sqrt(jnp.diag(K * self.y_std ** 2)),
+            )
+        else:
+            return (mean * self.y_std) + self.y_mean, K * self.y_std ** 2
+
+    def sample_y(self, rng_key, X):
+        # get posterior mean and covariance
+        y_mean, y_cov = self.predict(X)
+        # draw one sample
+        return jax.random.multivariate_normal(rng_key, mean=y_mean, cov=y_cov)
+
+
+# our TS-GP optimizer
+class ThompsonSamplingGP:
+
+    # initialization
+    def __init__(
+        self, gp, n_random_draws, objective, x_bounds, interval_resolution=1000
```

Comment:
instead of `interval_resolution` perhaps `num_thompson_samples` would be more informative?

Reply:
`interval_resolution` was indeed ambiguous. It is not the number of Thompson samples but rather the resolution of the grid. I clarified that in the code.